src/main/java/seedu/address/model/feed/Feed.java

```diff
@@ -31,52 +33,124 @@ public Feed(String name, String address) {
+    }
+
+    /**
+     * Fetches the feed's data as a String. This method first attempts to fetch data from the feed's remote
+     * address. If unsuccessful, it returns the cached feed data.
+     *
+     * @return String representing the feed data.
+     */
+    private String fetchFeedData() {
+        StringBuilder feedData = new StringBuilder();
+        int matchCount = 0;
 
         try {
+            // Fetch remote
             URL url = new URL(address);
             HttpURLConnection conn = (HttpURLConnection) url.openConnection();
             BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
 
-            int count = 0;
-            String title = null;
-            String link = null;
-            String buffer1;
-            String buffer2;
-
-            while ((buffer1 = in.readLine()) != null && count < 5) {
-                if (buffer1.contains("<title>")) {
-                    buffer2 = in.readLine();
-
-                    if (buffer2.contains("<link>")) {
-                        title = Jsoup.parse(buffer1).text()
-                                .replace("<title>", "").replace("</title>", "").strip();
-                        link = Jsoup.parse(buffer2).text()
-                                .replace("<link>", "").replace("</link>", "").strip();
-
-                        FeedPost feedPost = new FeedPost(name, title, link);
-                        posts.add(feedPost);
-                        count = count + 1;
-                    }
+            String titleBuffer = in.readLine();
+            String linkBuffer;
+
+            // Filter data
+            while (
+                    titleBuffer != null
+                            && (linkBuffer = in.readLine()) != null
+                            && matchCount < 5
+            ) {
+                titleBuffer = titleBuffer.strip();
+                linkBuffer = linkBuffer.strip();
+
+                if (titleBuffer.contains("<title>") && linkBuffer.contains("<link>")) {
+                    feedData.append(String.format("%s%n", titleBuffer));
+                    feedData.append(String.format("%s%n", linkBuffer));
+                    matchCount = matchCount + 1;
                 }
 
-                title = null;
-                link = null;
+                titleBuffer = linkBuffer;
             }
 
+            // Cleanup
             in.close();
             conn.disconnect();
+
+            LogsCenter.getLogger(Feed.class).info(
+                    String.format("[Feed: %s] Successfully fetched remote feed data", name));
         } catch (Exception e) {
-            LogsCenter.getLogger(Feed.class).warning("Failed to fetch posts from " + toString());
+            LogsCenter.getLogger(Feed.class).warning(
+                    String.format("[Feed: %s] Failed to fetch remote feed data, using local cache", name));
         }
 
-        return posts;
+
+        return feedData.length() == 0 ? null : feedData.toString();
```

Comment:
I remember my 1010 prof telling is a bad practice to use this instead of if else. Not very sure if he's right.